File: Python/Problems/Problems/problem1108.py
from .problem import Problem


class Problem1108(Problem):
    def test(self):
        address = "1.1.1.1"

        expected = "1[.]1[.]1[.]1"

        result = self.defangIPaddr(address)

        print(result)

        return expected == result

    def defangIPaddr(self, address):
        return address.replace(".", "[.]")

File: C++/Problems/Problems/problem_1108.cpp
#include "problem_1108.h"

#include <sstream>

bool problem_1108::test() {
    std::string address = "1.1.1.1";

    std::string expected = "1[.]1[.]1[.]1";

    auto result = defangIPaddr(address);

    std::cout << result << std::endl;

    return expected == result;
}

std::string problem_1108::defangIPaddr(std::string address) {
    std::ostringstream result;
    for (auto c : address) {
        if (c != '.')
            result << c;
        else
            result << "[.]";
    }
    return result.str();
}
